src/main/java/seedu/address/model/Concierge.java

```diff
@@ -165,6 +168,42 @@ public void setRooms(List<Room> rooms) {
         this.rooms.setRooms(rooms);
     }
 
+    /**
+     * Reassigns the booking identified by {@code startDate} in the room identified by {@code roomNumber} to the room
+     * identified by {@code newRoomNumber}.<br>
+     * The following checks are conducted in succession, each possibly throwing an exception:<br>
+     * If the booking does not exist, throw NoBookingFoundException.<br>
+     * If the booking is expired, throw ExpiredBookingException.<br>
+     * If room and new room are equal, throw OriginalRoomReassignException.<br>
+     * If new room has bookings and its first booking is already checked-in, throw ReassignToCheckedInRoomException.<br>
+     * If the booking overlaps with any of new room's bookings, throw OverlappingBookingException.
+     */
+    public void reassignRoom(RoomNumber roomNumber, LocalDate startDate, RoomNumber newRoomNumber) {
+        Room room = rooms.getRoom(roomNumber);
+        Room newRoom = rooms.getRoom(newRoomNumber);
+
+        Booking bookingToReassign = room.getBookings()
+                .getFirstBookingByPredicate(booking -> booking.getBookingPeriod().getStartDate().equals(startDate));
+
+        if (bookingToReassign.isExpired()) {
+            throw new ExpiredBookingException();
+        }
+
+        if (room.equals(newRoom)) {
+            throw new OriginalRoomReassignException();
+        }
+
+        if (newRoom.hasBookings() && newRoom.getBookings().getFirstBooking().getIsCheckedIn()) {
```

Comment:
What if the first booking is checked in, but the user hopes to reassign a booking from a later date?

For example, 
Room 1; Booking 1A (TOMORROW_NEXTWEEK)
Room 2; Booking 2A (YESTERDAY_TODAY), checked in

>> Would reassign Booking 1A to Room 2 throw an exception?

Reply:
Good catch! That should be allowed, but isn't. Will fix it!